pysilcam/tracking/silcamtracker.py

```diff
@@ -0,0 +1,410 @@
+from pysilcam.fakepymba import silcam_load
+import h5py
+import names
+
+
+class Tracker:
+    '''
+    Class for tracking
+    '''
+
+    def __init__(self):
+        # define some defaults
+        self.av_window = 15
+        self.THRESHOLD = 0.98
+        self.MIN_LENGTH = 500  # microns
+        self.MIN_SPEED = 0.01  # cm/s
+        self.GOOD_FIT = 0.2
+        self.PIX_SIZE = 27.532679738562095
+        self.ecd_tolerance = 0
+        self.path = ''
+        self.DATAFILE = ''
+        self.files = None
+        self.track_length_limit = 15
+
+    def initialise(self):
+        print('* INITIALISE')
+        if self.files is None:
+            self.files = [os.path.join(self.path, f)
+                          for f in sorted(os.listdir(self.path)) if
+                          f.endswith('.bmp') or f.endswith('.silc') or f.endswith('.silc_mono')]
+        print('  File list obtained:')
+        print('    ', len(self.files), 'files found')
+
+        self.aqgen = self.generator_tracker()
+
+        # Get number of images to use for background correction from config
+        print('* Initializing background image handler')
+        self.bggen = backgrounder(self.av_window, self.aqgen,
+                                  bad_lighting_limit=None,
+                                  real_time_stats=False)
+
+    def generator_tracker(self, datapath=None):
+        for f in tqdm(self.files):
+
+            img = silcam_load(f)
+
+            if np.ndim(img) == 3:
+                img = np.min(img, axis=2)
+            imx, imy = np.shape(img)
+            imc = np.zeros((imx, imy, 3), dtype=np.uint8())
+            imc[:, :, 0] = img
+            imc[:, :, 1] = img
+            imc[:, :, 2] = img
+            timestamp = pd.to_datetime(os.path.splitext(os.path.split(f)[-1])[0][1:])
+            yield timestamp, imc
+
+    def process(self):
+        PIX_SIZE = self.PIX_SIZE
+        MIN_LENGTH = self.MIN_LENGTH
+        GOOD_FIT = self.GOOD_FIT
+        DATAFILE = self.DATAFILE
+        track_length_limit = self.track_length_limit
+
+        if DATAFILE == '':
+            print('DATAFILE not specified')
+            print('---- END ----')
+            return
+
+        # setup HDF5 file and metadata
+        with h5py.File(DATAFILE + '.h5', "a") as HDF5File:
+            meta = HDF5File.require_group('Meta')
+            meta.attrs['Modified'] = str(pd.datetime.now())
+            meta.attrs['DatasetName'] = 'not implemented'
+            HDF5File.create_group("Tracking")
+
+        tracks = pd.DataFrame()
+        tracks.index.name = 'UPID'
+        UPID = -1
+
+        print('* Tracking....')
+
+        img1, t1 = self.load_image()
+        img2 = []
+        t2 = []
+
+        i = 0
+        while True:
+
+            if not (i == 0):
+                img1 = np.copy(img2)
+                t1 = pd.to_datetime(str(np.copy(t2)))
+
+            try:
+                img2, t2 = self.load_image()
+            except Exception as error:

```

Comment:
F841: local variable 'error' is assigned to but never used  <!-- Automatically posted by Lintly -->